src/solve.c

```diff
@@ -28,7 +28,30 @@
 int solve(INPUT *in)
 {
   LOG("solve CALLED", 2);
-  int toReturn = process_and(in);
+  int nbvars = in->nbvars;
+  //int toReturn = process_and(in);
+  //int * counts = (int*)malloc(sizeof(int)* nbvars);
+  int * setvals = (int*)malloc(sizeof(int)* nbvars);
+  int i;
+  for (i = 0; i < nbvars; i++)
+  {
+    //counts[i] = 0;
+    setvals[i] = -1;
+  }
+  int toReturn;
+  if (solver(in->data, nbvars, in->nbclauses, in->clause_lengths, in->value_sums, setvals, 0, 0))
+  {
+    toReturn = 1;
+  }
+  else
+  {
+    setvals[0] = !setvals[0];
+    toReturn = solver(in->data, nbvars, in->nbclauses, in->clause_lengths, in->value_sums, setvals, 0, 1);
+  }
+
+
+  //free(counts);
```

Comment:
Same as above, and it should be memory leak free, I just have a single malloc and a single free in the same method.